### PR TITLE
weather-forecast: Rewrote the story

### PR DESCRIPTION
**Pull request by MikaeelMF**
Hello!

Following issue #1579 I have tried to tweak the instructions.md document. I have added a background story and tried to make each task make sense in its context.

I would appreciate it if you could check it out and tell me if it needs any changes! 

Thank you!

### CODE REVIEW

**Comment by ErikSchierboom** [comment]
Thanks a lot for the PR! We've not yet found the time to review, but will do so as soon as we can.

**Review by kcpetersen111 — APPROVED** [review]
This looks good.  The instructions get the same meaning across in a bit clearer way with a bit of a story to go with it.

**Comment by MikaeelMF** [comment]
> This looks good. The instructions get the same meaning across in a bit clearer way with a bit of a story to go with it.

Is it okay to merge the request then?

**Comment by MikaeelMF** [comment]
> This looks good. The instructions get the same meaning across in a bit clearer way with a bit of a story to go with it.

Thank you for your comment!
Is it okay to merge the request then?

**Review by kcpetersen111 — APPROVED** [review]

**Review by kcpetersen111 — COMMENTED** [review]
Sorry I did not relize that someone with write access had to review it and not just to merge it. I dont have write access.

**Comment by ErikSchierboom** [comment]
CC @exercism/go

File: exercises/concept/weather-forecast/.docs/instructions.md [hunk 1]
```diff
@@ -1,21 +1,15 @@
 # Instructions
 
-Being hired by a big weather forecast company, your job is to maintain their code base. Scrolling through the code, you find it hard to follow what the code is actually doing. Good thing you know just the right thing to do!
+Goblinocus is a country that takes its weather forecast very seriously! Since you are a renowned responsible proficient developer, they have asked you to write a program for them that can forecast the current weather condition of various cities in Goblinocus. You were busy at that time and asked one of your friends to do the job instead. After a while, the president of Goblinocus contacted you and said they do not understand a bit about your friend's code! When you check the code, you find out your friend did not act as a responsible programmer and there are no comments in the code!!! Therefore as a responsible person, you feel obligated to clarify the program so goblins can understand them as well!
```
Comment (andrerfcsantos) [inline exercises/concept/weather-forecast/.docs/instructions.md:3]:
```suggestion
Goblinocus is a country that takes its weather forecast very seriously! Since you are a renowned responsible and proficient developer, they have asked you to write a program for them that can forecast the current weather condition of various cities in Goblinocus. You were busy at that time and asked one of your friends to do the job instead. After a while, the president of Goblinocus contacted you and said they do not understand a bit about your friend's code! When you check the code, you find out your friend did not act as a responsible programmer and there are no comments in the code! Therefore as a responsible person, you feel obligated to clarify the program so goblins can understand them as well!
```

Suggested by Grammarly.

File: exercises/concept/weather-forecast/.docs/instructions.md [hunk 2]
```diff
@@ -1,21 +1,15 @@
...
 ## 1. Document package weather
 
-Write a package comment for `package weather` that describes its contents. The package comment should introduce the package and provide information relevant to the package as a whole.
+Since goblins are not as smart as you are, they forgot what the package should do for them! Please write a comment for `package weather` that describes its contents. The package comment should introduce the package and provide information relevant to the package as a whole.
 
-## 2. Document the CurrentCondition variable
+## 2. Document the CurrentCondition and CurrentLocation variables
 
-Write a comment for the package variable `CurrentCondition`.
+The president of Goblinocus is a bit paranoid and fears uncommented variables are used for destroying their country! Please clarify the usage of the package variables `CurrentCondition` and `CurrentLocation` and put the president's mind at ease. This should tell any user of the package what information the variables store, and what they can do with it.
```
Comment (andrerfcsantos) [inline exercises/concept/weather-forecast/.docs/instructions.md:11]:
```suggestion
The president of Goblinocus is a bit paranoid and fears uncommented variables are used to destroy their country! Please clarify the usage of the package variables `CurrentCondition` and `CurrentLocation` and put the president's mind at ease. This should tell any user of the package what information the variables store, and what they can do with it.
```

Suggested by Grammarly.

File: exercises/concept/weather-forecast/.docs/instructions.md [hunk 3]
```diff
@@ -1,21 +1,15 @@
...
 
 ## 1. Document package weather
 
-Write a package comment for `package weather` that describes its contents. The package comment should introduce the package and provide information relevant to the package as a whole.
+Since goblins are not as smart as you are, they forgot what the package should do for them! Please write a comment for `package weather` that describes its contents. The package comment should introduce the package and provide information relevant to the package as a whole.
 
-## 2. Document the CurrentCondition variable
+## 2. Document the CurrentCondition and CurrentLocation variables
```
Comment (andrerfcsantos) [inline exercises/concept/weather-forecast/.docs/instructions.md:9]: Changes to the instructions' titles must also be reflected in the [hints file](https://github.com/exercism/go/blob/main/exercises/concept/weather-forecast/.docs/hints.md). Can you please also apply this change there as well?

File: exercises/concept/weather-forecast/.docs/instructions.md [hunk 4]
```diff
@@ -1,21 +1,15 @@
...
 
-Write a comment for the package variable `CurrentCondition`.
+The president of Goblinocus is a bit paranoid and fears uncommented variables are used for destroying their country! Please clarify the usage of the package variables `CurrentCondition` and `CurrentLocation` and put the president's mind at ease. This should tell any user of the package what information the variables store, and what they can do with it.
 
-This should tell any user of the package what information the variable stores, and what they can do with it.
+## 3. Document the Forecast() function
```
Comment (andrerfcsantos) [inline exercises/concept/weather-forecast/.docs/instructions.md:13]: Changes to the instructions' titles must also be reflected in the [hints file](https://github.com/exercism/go/blob/main/exercises/concept/weather-forecast/.docs/hints.md). Can you please also apply this change there as well?

**Review by andrerfcsantos — CHANGES_REQUESTED** [review]
Thanks a lot for this and sorry for the delay in reviewing it.

The story looks way better indeed!

I just ran the copy through Grammarly and made some suggestions based on its suggestions. The titles in the hints files also must be changed.

Other than that looks great and we can merge after that :)

**Review by andrerfcsantos — APPROVED** [review]
Awesome!

Sorry for the delay in reviewing this.
